src/bin/warg.rs

```diff
@@ -46,27 +46,98 @@ async fn main() -> Result<()> {
         .with_env_filter(EnvFilter::from_default_env())
         .init();
 
-    if let Err(e) = match WargCli::parse() {
-        WargCli::Config(cmd) => cmd.exec().await,
-        WargCli::Info(cmd) => cmd.exec().await,
-        WargCli::Key(cmd) => cmd.exec().await,
-        WargCli::Lock(cmd) => cmd.exec(None).await,
-        WargCli::Bundle(cmd) => cmd.exec(None).await,
-        WargCli::Dependencies(cmd) => cmd.exec(None).await,
-        WargCli::Download(cmd) => cmd.exec(None).await,
-        WargCli::Update(cmd) => cmd.exec(None).await,
-        WargCli::Publish(cmd) => cmd.exec(None).await,
-        WargCli::Reset(cmd) => cmd.exec().await,
-        WargCli::Clear(cmd) => cmd.exec().await,
-        WargCli::Login(cmd) => cmd.exec().await,
-        WargCli::Logout(cmd) => cmd.exec().await,
-    } {
-        if let Some(e) = e.downcast_ref::<ClientError>() {
-            describe_client_error_or_retry(e).await?;
-        } else {
-            eprintln!("error: {e:?}");
+    match &WargCli::parse() {
+        WargCli::Config(cmd) => cmd.clone().exec().await?,
+        WargCli::Info(cmd) => cmd.clone().exec().await?,
+        WargCli::Key(cmd) => cmd.clone().exec().await?,
+        WargCli::Lock(cmd) => {
+            with_interactive_retry(|retry: Option<Retry>| async {
```

Comment:
Can this go outside the `match`? The commands that don't support `retry` shouldn't be returning this error anyway and can just ignore the `retry` variable?